### PR TITLE
Add support for VARBINARY input to approx_distinct Presto aggregate function

### PR DESCRIPTION
**Pull request by zhli1142015**

### CODE REVIEW

**Comment by zhli1142015** [comment]
@mbasmanova and @Yuhta , could you help to take a look this change? Thanks.
Here is the background for this PR:
We map Spark agg function `HyperLogLogPlusPlus` to this func in Gluten. By Spark UT: https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala#L224C19-L224C29, we found this function is not registered for `varbinary` type.

**Comment by mbasmanova** [comment]
@zhli1142015 CI is red. Please, take a look.

**Comment by jinchengchenghh** [comment]
an you add a test?

File: velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp [hunk 1]
```diff
@@ -230,6 +230,17 @@ TEST_F(ApproxDistinctTest, globalAggStrings) {
   testGlobalAgg(values, kFruits.size());
 }
 
+TEST_F(ApproxDistinctTest, globalAggVarbinary) {
+  vector_size_t size = 1'000;
```
Comment (jinchengchenghh) [inline velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp:234]: No need to initialize size since it is used only once

**Comment by zhli1142015** [comment]
> Can you add a test?

Added UT, thanks.

**Review by jinchengchenghh — APPROVED** [review]

**Comment by zhli1142015** [comment]
> @zhli1142015 CI is red. Please, take a look.

CI passed, @mbasmanova , could you help review?
Thanks.

**Comment by zhli1142015** [comment]
Kindly ping, @mbasmanova could you help to review? Thanks.

File: velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp [hunk 2]
```diff
@@ -504,7 +505,7 @@ exec::AggregateRegistrationResult registerApproxDistinct(
           const TypePtr& resultType,
           const core::QueryConfig& /*config*/)
           -> std::unique_ptr<exec::Aggregate> {
-        TypePtr type = argTypes[0]->isVarbinary() ? BIGINT() : argTypes[0];
+        TypePtr type = hllAsRawInput ? BIGINT() : argTypes[0];
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp:508]: I'm not sure I understand this logic.

It looks like we only use 'type->kind()' and it is only used when hllAsRawInput is false. Hence, it seems we can simply use argTypes[0] for type.

```
const TypePtr& type = argTypes[0];
```

**Review by mbasmanova — COMMENTED** [review]
@zhli1142015 Some comments.

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Pull request closed without merging**